### DISCUSSION TITLE
"@antv/s2": "^2.1.5"和"@antv/s2-react": "^2.1.4"，无法正常使用

### DISCUSSION THREAD

**Discussion by gb853940223** [antvis/S2#3054] [Q&A] [answered]
创建了一个全新的React Demo，安装了最新版的@antv/s2和@antv/s2-react。使用时报了如下错误：
![image](https://github.com/user-attachments/assets/5359cba0-033d-4f0e-b78f-038bee59dafe)
这个是pagage.json
![image](https://github.com/user-attachments/assets/05eeb79c-8530-48e2-b430-d059ed1156e7)
这个是使用方式
![image](https://github.com/user-attachments/assets/1a4fa48f-0669-4753-ab26-13fd0fcc60a4)
是使用问题还是包本身的问题？
有没有遇到和我类似问题的大佬？

**Comment by petercat-assistant** [comment]
目前没有找到与您描述的问题直接相关的已知问题或讨论。您遇到的问题可能是由于某些特定的配置或使用方式导致的。

为了帮助您更好地解决问题，请您检查以下几点：
1. 确保 `@antv/s2` 和 `@antv/s2-react` 的版本是兼容的。
2. 检查项目中的其他依赖项是否与 `@antv/s2` 和 `@antv/s2-react` 存在版本冲突。
3. 查看控制台的错误信息，可能会提供更具体的错误原因。
4. 尝试在一个全新的项目中仅安装 `@antv/s2` 和 `@antv/s2-react`，以排除其他依赖项的影响。

如果您仍然无法解决问题，建议您在 GitHub 上创建一个新的 issue，提供详细的错误信息和复现步骤，以便开发团队能够更好地帮助您。

<details>
<summary>🪧 Tips</summary>
For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
</details>

**Comment by lijinke666** [comment, marked as answer]
确保你的打包工具识别 `package.json` 的 `exports` 和 `peerDependenciesMeta` 关键字, `@antv/g2` 是可选依赖, 只有使用 `@antv/s2/extends` 才需要, 看起来是你的打包工具不支持, 实在不行可以额外安装下 `@antv/g2`

```ts
  "exports": {
    ".": {
      "import": "./esm/index.js",
      "require": "./lib/index.js"
    },
    "./extends": {
      "import": "./esm/extends/index.js",
      "require": "./lib/extends/index.js"
    },
    "./*": "./*"
  },
  "peerDependencies": {
    "@antv/g2": ">=5.1.21"
  },
  "peerDependenciesMeta": {
    "@antv/g2": {
      "optional": true
    }
  },
```

**Comment by mooninshoe** [comment]


**Reply by Alexzjt** [reply]
先安装G2使用吧
